### PR TITLE
Migrate metadata from setup.py to setup.cfg

### PR DESCRIPTION
**Pull request by cclauss**
Created using https://pypi.org/project/setuptools-py2cfg plus manual modifications.

https://setuptools.pypa.io/en/latest/userguide/quickstart.html#basic-use
> We recommend users expose as much as possible configuration in a more declarative way via the [pyproject.toml](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html) or [setup.cfg](https://setuptools.pypa.io/en/latest/userguide/declarative_config.html), and keep the setup.py minimal with only the dynamic parts (or even omit it completely if applicable).

https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html
https://thehackernews.com/2022/09/warning-pypi-feature-executes-code.html

### CODE REVIEW

**Comment by sigmavirus24** [comment]
@cclauss would you be willing to resolve the conflicts here?

**Comment by cclauss** [comment]
Should `flake8_import_order/__about__.py` be removed?

**Comment by sigmavirus24** [comment]
Could we not use `attr: flake8_import_order.__about__.__version__` in `setup.cfg` to avoid the version being hard-coded in `setup.cfg`? Also the version in there now is wrong.

File: flake8_import_order/__about__.py [hunk 1]
```diff
@@ -1,24 +0,0 @@
-from __future__ import absolute_import, division, print_function
```
Comment (sigmavirus24) [inline flake8_import_order/__about__.py:1]: I don't think this should still be deleted, right?

Comment (cclauss) [reply flake8_import_order/__about__.py]: These `__future__` imports can be dropped now that Python 2 has been dropped.
* https://docs.python.org/3/library/__future__.html

`pyupgrade` will automate that process.

**Review by sigmavirus24 — APPROVED** [review]

**Pull request merged**